src/plugin/PluginLoadabilityChecker.php

```diff
@@ -71,6 +72,12 @@ public function check(PluginDescription $description) : Translatable|null{
 			return KnownTranslationFactory::pocketmine_plugin_incompatibleOS(implode(", ", $description->getCompatibleOperatingSystems()));
 		}
 
+		$pluginPhpVersions = $description->getCompatiblePhpVersions();
+		$pluginCompatiblePhpVersions = array_filter($pluginPhpVersions, static fn(string $version) => version_compare(phpversion(), $version, 'le'));
+		if(count($pluginPhpVersions) > 0 && count($pluginCompatiblePhpVersions) < 1){
+			return KnownTranslationFactory::pocketmine_plugin_incompatiblePhpVersion(implode(", ", $pluginPhpVersions));
+		}
+
```

Comment:
I thought the consensus was to use the same system as extensions, for consistency. Some of the code below might need to be extracted into a separate function, unless you get creative.

Reply:
Do you mean I should add 'Core' to the required extensions list instead of filtering it separately here?

Reply:
I mean that the `php` directive should have the same functionality as the `extensions.Core` directive.